src/kubernetes_cluster/spec/cluster.rs

```diff
@@ -385,63 +395,48 @@ pub open spec fn stutter<K: ResourceView, E: ExternalAPI, R: Reconciler<K, E>>()
+pub open spec fn next_step(s: State<K, E, R>, s_prime: State<K, E, R>, step: Step<K, E>) -> bool {
     match step {
-        Step::KubernetesAPIStep(input) => kubernetes_api_next().forward(input)(s, s_prime),
-        Step::ControllerStep(input) => controller_next::<K, E, R>().forward(input)(s, s_prime),
-        Step::ClientStep(input) => client_next().forward(input)(s, s_prime),
-        Step::ExternalAPIStep(input) => external_api_next().forward(input)(s, s_prime),
-        Step::ScheduleControllerReconcileStep(input) => schedule_controller_reconcile().forward(input)(s, s_prime),
-        Step::RestartController() => restart_controller().forward(())(s, s_prime),
-        Step::DisableCrash() => disable_crash().forward(())(s, s_prime),
-        Step::KubernetesBusy(input) => busy_kubernetes_api_rejects_request().forward(input)(s, s_prime),
-        Step::DisableBusy() => disable_busy().forward(())(s, s_prime),
-        Step::StutterStep() => stutter().forward(())(s, s_prime),
+        Step::KubernetesAPIStep(input) => Self::kubernetes_api_next().forward(input)(s, s_prime),
+        Step::ControllerStep(input) => Self::controller_next().forward(input)(s, s_prime),
+        Step::ClientStep(input) => Self::client_next().forward(input)(s, s_prime),
+        Step::ExternalAPIStep(input) => Self::external_api_next().forward(input)(s, s_prime),
+        Step::ScheduleControllerReconcileStep(input) => Self::schedule_controller_reconcile().forward(input)(s, s_prime),
+        Step::RestartController() => Self::restart_controller().forward(())(s, s_prime),
+        Step::DisableCrash() => Self::disable_crash().forward(())(s, s_prime),
+        Step::KubernetesBusy(input) => Self::busy_kubernetes_api_rejects_request().forward(input)(s, s_prime),
+        Step::DisableBusy() => Self::disable_busy().forward(())(s, s_prime),
+        Step::StutterStep() => Self::stutter().forward(())(s, s_prime),
     }
 }
 
 /// `next` chooses:
 /// * which host to take the next action (`Step`)
 /// * whether to deliver a message and which message to deliver (`Option<Message>` in `Step`)
-pub open spec fn next<K: ResourceView, E: ExternalAPI, R: Reconciler<K, E>>() -> ActionPred<State<K, E, R>> {
-    |s: State<K, E, R>, s_prime: State<K, E, R>| exists |step: Step<K, E>| next_step::<K, E, R>(s, s_prime, step)
+pub open spec fn next() -> ActionPred<State<K, E, R>> {
+    |s: State<K, E, R>, s_prime: State<K, E, R>| exists |step: Step<K, E>| Self::next_step(s, s_prime, step)
 }
 
 /// We install the reconciler to the Kubernetes cluster state machine spec
 /// TODO: develop a struct for the compound state machine and make reconciler its member
 /// so that we don't have to pass reconciler to init and next in the proof.
-pub open spec fn sm_spec<K: ResourceView, E: ExternalAPI, R: Reconciler<K, E>>() -> TempPred<State<K, E, R>> {
-    lift_state(init::<K, E, R>()).and(sm_partial_spec::<K, E, R>())
+pub open spec fn sm_spec() -> TempPred<State<K, E, R>> {
```

Comment:
I think we probably no longer need `sm_spec` and `sm_partial_spec`, but let's keep them for now